app/connectors/runner.py

```
@@ -0,0 +1,138 @@
+
+    async def _run_connector(self, connector: IConnector) -> None:
+        """Run one connector: start, consume events(), ingest; restart on crash with backoff."""
+        backoff_seconds = 5
+        max_backoff = 300
+        while True:
+            try:
+                await connector.start()
+                it: AsyncIterator[list[ConnectorEvent]] = connector.events()
+                async for batch in it:
+                    await self._ingest_batch(connector, batch)
+                return  # clean exit
```

Ruff (0.15.5)

[warning] Consider moving this statement to an `else` block

(TRY300)